### ISSUE TITLE
Using function keys for hotkeys (like F5 to run query) in SQLPro for MSSQL

### ISSUE THREAD

**Issue by aaksyonenko** [hankinsoft/SQLPro#426]
Feature request.
F5 is a default macOS 'display autocomplete' shortcut (see https://osxdaily.com/2011/04/21/word-completion-mac-os-x/).  However, for those of us coming from the SSIS background, it is also the default SSIS shortcut to execute the selected SQL.
It would be great to have SQLPro intercept that before it gets to macOS and use it as a custom hotkey (among other possible function keys and macOS defaults).
Optimal solution would be to default everything to macOS unless a custom hotkey has been defined in SQLPro to override whatever it may be (not limited to F5) - I personally like this one better than the below alternative.
Or, alternatively, use a modifier key, for example "F5" defaults to macOS, but Shift+F5 (or whatever) gets intercepted - that way Mac users still have their familiar functionality and Windows SSIS users have a minimal learning curve.

**Comment by hankinsoft** [comment]
Fixed for the next build. I forget, are you using the website version of the app, or the Mac app store version?

**Comment by aaksyonenko** [comment]
Website

**Comment by hankinsoft** [comment]
Can you grab https://d3fwkemdw8spx3.cloudfront.net/mssql/SQLProMSSQL.2020.47.app.zip and give it a try?

**Comment by aaksyonenko** [comment]
Didn't work :-(  
P.S. I went the whole 9 yards and purchased the full Studio, so if this fix is at all possible - would be great to have it there, too :-)
![2020-05-20_10-38-01](https://user-images.githubusercontent.com/22381393/82459842-66403300-9a86-11ea-8198-69d0097ea693.png)
![2020-05-20_10-38-42](https://user-images.githubusercontent.com/22381393/82459857-6b9d7d80-9a86-11ea-8348-17732b2fb858.png)

**Comment by hankinsoft** [comment]
Sure, I'll get the fix in studio as well. A few questions:

1. If you close the app and re-open (while still leaving F5 as the default), does that fix it?
2. Are you using an external keyboard? Odd question, but could I get a picture of your keyboard?
3. If you press the 'Data' menu, does the shortcut button show F5?
4. In the query tab, does the 'Run all/Run selected' button show F5 as the shortcut key?

**Comment by aaksyonenko** [comment]
1. closing and re-opening didn't help
2. Yes, I am using external keyboard: https://www.daskeyboard.com/daskeyboard-4-professional-for-mac/
3. Data menu shows "run query F" (no 5)
4. The "run all F5" on top right shows "F5"

**Comment by hankinsoft** [comment]
1. I like your keyboard. I have a [CODE](https://codekeyboards.com) keyboard which I like, but wish it had USB slots. The usb-micro port on it is starting to go bad, so I'm going to be in the market for a new keyboard soon.

2. Can you try this build: https://d3fwkemdw8spx3.cloudfront.net/studio/SQLProStudio.2020.47.app.zip

That should sort it. Please let me know!

**Comment by aaksyonenko** [comment]
Works perfectly, thank you!

Thanks.  Yeah, I found it when I was looking for a full-size mechanical keyboard with NKRO (I hate not being able to hear myself type).  Large hands and fast typing are not at all compatible with the Magic Keyboard :-( What sold me initially was the volume control on the board.  But I have seen the value of additional USB ports since then — an MBP definitely doesn't have enough of those.  I got two additional hubs for all my stuff.
